core/services/ocr2/plugins/ocr2keeper/evm21/streams_lookup.go

```diff
@@ -499,20 +501,46 @@ func (r *EvmRegistry) multiFeedsRequest(ctx context.Context, ch chan<- MercuryDa
 			if resp.StatusCode == http.StatusUnauthorized {
 				retryable = false
 				state = encoding.UpkeepNotAuthorized
-				return fmt.Errorf("at timestamp %s upkeep %s received status code %d from mercury v0.3, most likely this is caused by unauthorized upkeep", sl.Time.String(), sl.upkeepId.String(), resp.StatusCode)
+				return fmt.Errorf("at timestamp %s upkeep %s received status code %d from mercury v0.3, most likely this upkeep does not have access to certain feed IDs", sl.Time.String(), sl.upkeepId.String(), resp.StatusCode)
 			} else if resp.StatusCode == http.StatusBadRequest {
 				retryable = false
 				state = encoding.InvalidMercuryRequest
-				return fmt.Errorf("at timestamp %s upkeep %s received status code %d from mercury v0.3 with message: %s", sl.Time.String(), sl.upkeepId.String(), resp.StatusCode, string(body))
+				return fmt.Errorf("at timestamp %s upkeep %s received status code %d from mercury v0.3, most likely the requested feed ids are not properly hex encoded", sl.Time.String(), sl.upkeepId.String(), resp.StatusCode)
 			} else if resp.StatusCode == http.StatusInternalServerError {
 				retryable = true
 				state = encoding.MercuryFlakyFailure
 				return fmt.Errorf("%d", http.StatusInternalServerError)
+			} else if resp.StatusCode == http.StatusUnprocessableEntity {
+				retryable = false
+				state = encoding.InvalidMercuryRequest
+				return fmt.Errorf("at timestamp %s upkeep %s received status code %d from mercury v0.3, most likely feed ID exists, user has permissions, but the feed is not active (deactivated). This should not happen. Contact Mercury team about this", sl.Time.String(), sl.upkeepId.String(), resp.StatusCode)
 			} else if resp.StatusCode == 420 {
 				// in 0.3, this will happen when missing/malformed query args, missing or bad required headers, non-existent feeds, or no permissions for feeds
 				retryable = false
 				state = encoding.InvalidMercuryRequest
 				return fmt.Errorf("at timestamp %s upkeep %s received status code %d from mercury v0.3, most likely this is caused by missing/malformed query args, missing or bad required headers, non-existent feeds, or no permissions for feeds", sl.Time.String(), sl.upkeepId.String(), resp.StatusCode)
+			} else if resp.StatusCode == http.StatusPartialContent {
+				retryable = true
+				state = encoding.MercuryFlakyFailure
+
+				// request is completely valid but mercury server has not generated reports for some feeds
+				var response MercuryV03Response
+				err1 = json.Unmarshal(body, &response)
+				if err1 != nil {
+					lggr.Warnf("at timestamp %s upkeep %s failed to unmarshal body to MercuryV03Response from mercury v0.3: %v", sl.Time.String(), sl.upkeepId.String(), err1)
+					retryable = false
+					state = encoding.MercuryUnmarshalError
+					return err1
+				}
+				var missingFeeds []string
+				for _, rsp := range response.Reports {
+					if rsp.Error == dataNotFound {
```

Comment:
I'm not sure if it'd be a good idea to check the error message explicitly - what if we provided an Error code of sorts?I was thinking something similar to [this](https://developers.google.com/maps/documentation/geocoding/requests-geocoding#StatusCodes) I found that Google does this for some APIs. Providing this would allow you to be able to only do this for certain error codes that are less likely to change than the error message - because we may end up adding like an "Error", "ErrorDetail"